### ISSUE TITLE
Split out directory reading into its own class

### ISSUE THREAD

**Issue by lcabrini** [lcabrini/jfm#23]
To keep the code as simple as possible, let's split out the directory reading code into it's own class. Later we are going to have to do things like MIME-type detection in order to render the correct icon for each entry as well as offer different options depending on MIME-type.

**Comment by CollinsLord** [comment]
Please i want to know which issue should i be targeting now,looking at things most of the issues has been address in the previous pull request .

**Comment by lcabrini** [comment]
Yes, I need to sort everything out a bit. It's a bit problematic because of the way the commits were done. I don't get back references. In the future, please try to commit each issue separately so that we avoid this.

I'll go through and try to close issues and then create new ones.

**Comment by CollinsLord** [comment]
Okay. Thanks 

_Sent from my TECNO-C9 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by CollinsLord** [comment]
Please am waiting on you to start working on JFM
